pyproject.toml

```diff
@@ -0,0 +1,262 @@
+
+[tool.ruff.lint.mccabe]
+max-complexity = 15
+
+[tool.ruff.lint]
+ignore = [
+	# indentation contains tabs
+	"W191",
+]
+logger-objects = ["logHandler.log"]
+
+[tool.ruff.lint.per-file-ignores]
+# sconscripts contains many inbuilt functions not recognised by the lint,
+# so ignore F821.
+"sconstruct" = ["F821"]
+"*sconscript" = ["F821"]
+
+[tool.ruff.lint.flake8-tidy-imports.banned-api]
+"winsound.PlaySound".msg= "PlaySound uses winmm, which is deprecated in favor of the Windows core audio APIs."
+
+[tool.licensecheck]
+only_licenses = ["BSD", "MIT", "Python", "LGPLV3+", "Apache"]
+ignore_packages = [
+	# Compatible licenses:
+	"certifi",  # Mozilla Public License 2.0
+	"markdown-link-attr-modifier",  # GPLV3 license, but not in PyPI correctly
+	"pycaw",  # MIT license, but not in PyPI
+	"urllib3",  # MIT license, but not in PyPI
+	"wxPython",  # wxWindows Library License
+]
+
+
+[tool.pyright]
+venvPath = ".venv"
+venv = "."
+pythonPlatform = "Windows"
+typeCheckingMode = "strict"
+
+include = [
+	"**/*.py",
+	"**/*.pyw",
+]
+
+exclude = [
+	"sconstruct",
+	"*sconscript",
+	".git",
+	"__pycache__",
+	".venv",
+	# When excluding concrete paths relative to a directory,
+	# not matching multiple folders by name e.g. `__pycache__`,
+	# paths are relative to the configuration file.
+]
+
+# Tell pyright where to load python code from
+extraPaths = [
+	"./addon",
+]
+
+# General config
+analyzeUnannotatedFunctions = true
+deprecateTypingAliases = true
+
+# Stricter typing
+strictParameterNoneValue = true
+strictListInference = true
+strictDictionaryInference = true
+strictSetInference = true
+
+# Compliant rules
+reportAssertAlwaysTrue = true
+reportAssertTypeFailure = true
+reportDuplicateImport = true
+reportIncompleteStub = true
+reportInconsistentOverload = true
+reportInconsistentConstructor = true
+reportInvalidStringEscapeSequence = true
+reportInvalidStubStatement = true
+reportInvalidTypeVarUse = true
+reportMatchNotExhaustive = true
+reportMissingModuleSource = true
+reportMissingImports = false # temporarily setting to false for MathCAT integration
+reportNoOverloadImplementation = true
+reportOptionalContextManager = true
+reportOverlappingOverload = true
+reportPrivateImportUsage = true
+reportPropertyTypeMismatch = true
+reportSelfClsParameterName = true
+reportShadowedImports = true
+reportTypeCommentUsage = true
+reportTypedDictNotRequiredAccess = true
+reportUndefinedVariable = true
+reportUnusedExpression = true
+reportUnboundVariable = true
+reportUnhashable = true
+reportUnnecessaryCast = true
+reportUnnecessaryContains = true
+reportUnnecessaryTypeIgnoreComment = true
+reportUntypedClassDecorator = true
+reportUntypedFunctionDecorator = true
+reportUnusedClass = true
+reportUnusedCoroutine = true
+reportUnusedExcept = true
+
+# Should switch to true when possible
+reportDeprecated = false  # 1834 errors
+
+# Can be enabled by generating type stubs for modules via pyright CLI
+reportMissingTypeStubs = false
+
+# Bad rules
```

Comment:
it's probably worth clearing up all these comments about counting errors. They're only relevant to NVDA